src/Func.cpp

```diff
@@ -241,8 +253,8 @@ int Func::add_implicit_vars(vector<Expr> &args) const {
 }
 
 namespace {
-bool var_name_match(string candidate, string var) {
-    internal_assert(var.find('.') == string::npos)
+bool var_name_match(string candidate, string var, bool assert_on=true) {
```

Comment:
This looks fishy. In the cases where you call this with assert_on = false, is it not enough to test for string equality?